### ISSUE TITLE
Segmentation fault in skl_rngH_ng

### ISSUE THREAD

**Issue by pvanheus** [ogotoh/spaln#15]
Hi there

For some of my data (a minority of it), spaln is dumping core in `skl_rngH_ng`. The data for this is in Google Drive: https://drive.google.com/open?id=1-7gX8k76XBEtvDLfyZLz755VE-n_RMZH

Apologies for the odd file naming - this was originally run under Galaxy. The command was:

```
ALN_DBS=$(pwd)/dataset_11326_files
spaln -t2 -O0 -o output1 -Q7 -TCynoglossus -yS100.0 -dspalndb dataset_9742.dat
```

The database was created from the file `hard_masked_seabass_genome.fasta` (also in the Google Drive)

Debugging the crash reveals that it happens at https://github.com/ogotoh/spaln/blob/master/src/fwd2h1.c#L705 when dereferencing `cs`. Here is some relevant code:

```
VTYPE skl_rngH_ng(Seq* seqs[], Gsinfo* gsi, PwdB* pwd)
{
static	const char*	fswarn = "%s %s FrameShift: %d %d\n";
	Seq*	a = seqs[0];
	Seq*	b = seqs[1];
	SKL*	wsk = gsi->skl;
	int	num = (wsk++)->n;
	SpJunc*	spjcs = new SpJunc(b);
	VTYPE	h = 0;		// 
	VTYPE	hi = NEVSEL;		// intron-containing path
	VTYPE	ha = 0;
	VTYPE	hb = 0;
	VTYPE	hvl = 0;
	bool	ivl = false;
	VTYPE	sig5 = 0;
	VTYPE	sig3 = 0;
	int	insert = 0;
	int	deletn = 0;
	int	intlen = 0;
	int	preint = 0;
	int	phs = 0;
	int	nb = 0;
	int	maxexon = IntronPrm.rlmt;
	EISCR	rbuf;
	FSTAT*	fst = &gsi->fstat;
	FSTAT	pst;
	Eijnc*	eijnc = gsi->eijnc = new Eijnc();
	Cigar*	cigar = gsi->cigar = 0;
	Vulgar*	vlgar = gsi->vlgar = 0;
	switch (algmode.nsa) {
	    case EXN_FORM: 
	    case CIG_FORM: cigar = gsi->cigar = new Cigar(); break;
	    case VLG_FORM: vlgar = gsi->vlgar = new Vulgar(); break;
	    default: break;
	}
	vclear(fst);
	vclear(&pst);
	vclear(&rbuf);
	gsi->noeij = 0;
	CHAR*	cs = b->at(wsk[num-1].n - 2);
        int	termcodon = (*cs == TRM || *cs == TRM2);
```
Inspection with gdb shows that `num` is 48, and `wsk[num-1]` is: `{m = -2147483648, n = -2147483648}` which seems very suspicious. A bit more inspection shows that `wsk[num-2]` is `{m = 715794017, n = -2147483648}` and `wsk[num-3]` is `{m = 1125, n = 104970}` suggesting that there is something wrong in the `wsk` array.

Looking at the `sname` fields of sequences `a` and `b`, they appear to be `XP_016100957.1` and `NW_017363881.1` respectively, the first of which is a 1200 amino acid protein and the second of which is a 4679640 nucleotide stretch of DNA. 

That is as far as my debugging has been able to take for now.

**Comment by pvanheus** [comment]
A further note - if `XP_016100957.1` alone is used as a query, no segmentation fault happens.

**Comment by pvanheus** [comment]
This appears to be fixed in spaln 2.3.3c.
